rust-runtime/aws-smithy-types/src/timeout.rs

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! This module defines types that describe timeouts for the various stages of an HTTP request.
+
+use std::borrow::Cow;
+use std::fmt::Display;
+use std::time::Duration;
+
+/// Configuration for the various kinds of timeouts supported by aws_smithy_client::Client.
```

Comment:
I think this makes sense from a Smithy perspective, but I'm wondering if there's a way to phrase it so that it makes sense for both Smithy and the AWS SDK. That might just be something short like, "Operation timeout configuration".

Reply:
What about
> A container of optional [`Duration`]s for all of the supported AWS timeouts.

Reply:
Smithy is intended to be used outside of AWS, so I would drop that from it. I'm not sure that adding more verbage is necessarily adding value here. I think "timeout configuration" is probably as informative as "a container of optional durations ... for timeouts", but with less words.